pkg/config/system_probe.go

```diff
@@ -224,6 +224,7 @@ func InitSystemProbeConfig(cfg Config) {
 	cfg.BindEnv(join(netNS, "max_http_stats_buffered"), "DD_SYSTEM_PROBE_NETWORK_MAX_HTTP_STATS_BUFFERED")
 	cfg.BindEnv(join(smNS, "max_http_stats_buffered"))
 	cfg.BindEnvAndSetDefault(join(smNS, "max_kafka_stats_buffered"), 100000)
+	cfg.BindEnvAndSetDefault(join(smNS, "max_concurrent_requests"), 65536)
```

Comment:
I think we should use `AdjustUSM` to set the default value to MaxTracedConnections

Reply:
can you provide more context about that?

Reply:
AdjustUSM method allows us to put more logic in the configuration setup, deprecate values, set sophisticated checks and limitations, and also use other configuration value as a default value for our configuration key
the following examples checks if our new field is not set, or set with a zero value, in such a case, we put the default value to be the actual values of `max_tracked_connections`
```go
	if !cfg.IsSet(smNS("max_concurrent_requests")) || cfg.GetInt(smNS("max_concurrent_requests")) == 0 {
		cfg.Set(smNS("max_concurrent_requests"), cfg.GetInt(spNS("max_tracked_connections")))
	}
```

Thus if `max_tracked_connections` is set to 130k, our default matches with them (rather than putting the default value of `max_tracked_connections`)

Reply:
you're still not matching to max tracked connections by default
on prod and staging it is overriden to 130K

Reply:
OK, I understand it now. Added a unit test to assert the expected behavior as well.